custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js

```diff
@@ -104,10 +104,21 @@ function MapOrSectorController($location, storageService, locationsService) {
                         var location = locations[0];
                         $location.search('location_name', location.name);
                         $location.search('location_id', location.location_id);
-
                         storageService.setKey('search', $location.search());
                         if (location.location_type_name === 'awc') {
-                            $location.path('awc_reports');
+                            var awcReportPath = 'awc_reports';
+                            if ($location.path().indexOf('maternal_child') !== -1 || $location.path().indexOf('maternal_and_child') !== -1) {
```

Comment:
Not blocking, but I think it's possible this would be a little more comprehensible (and be slightly more DRY) if you defined a data structure like this:

```javascript
{
  'maternal_child': 'maternal_child',
  'maternal_and_child': 'maternal_child',
  'demographics': 'demographics',
  // etc.
}
```

And then updated the path in a loop.

Reply:
Updated, Please review